File: frontend/src/types/institution.ts
export interface Institution {
  id: string
  name: string
  location: string | null
  contact_name: string | null
  contact_mobile: string | null
  contact_wechat: string | null
}

export interface InstitutionCreate {
  name: string
  location?: string
  contact_name?: string
  contact_mobile?: string
  contact_wechat?: string
}

export interface InstitutionUpdate {
  name?: string
  location?: string
  contact_name?: string
  contact_mobile?: string
  contact_wechat?: string
}

File: backend/app/schemas/institution.py
import uuid
from datetime import datetime
from pydantic import BaseModel, Field


class InstitutionCreate(BaseModel):
    name: str = Field(..., max_length=50)
    location: str | None = Field(None, max_length=100)
    contact_name: str | None = Field(None, max_length=30)
    contact_mobile: str | None = Field(None, max_length=11)
    contact_wechat: str | None = Field(None, max_length=30)


class InstitutionUpdate(BaseModel):
    name: str | None = Field(None, max_length=50)
    location: str | None = Field(None, max_length=100)
    contact_name: str | None = Field(None, max_length=30)
    contact_mobile: str | None = Field(None, max_length=11)
    contact_wechat: str | None = Field(None, max_length=30)


class InstitutionRead(BaseModel):
    id: uuid.UUID
    name: str
    location: str | None
    contact_name: str | None
    contact_mobile: str | None
    contact_wechat: str | None

    model_config = {"from_attributes": True}